omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,98 @@
+        "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*" })
+public class AutoGenerationOptionResource extends DelegatingCrudResource<AutoGenerationOption> {
+  @Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = null;
+		
+		// NOTE: description for ref representation is provided by MetaDataDelegatingCrudResource
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+      description = new DelegatingResourceDescription();
+      
+			description.addProperty("identifierType");
+			description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+		}
+		
+		return description;
+  }
+  
+  @Override
+  public DelegatingResourceDescription getCreatableProperties() {
+      DelegatingResourceDescription description = new DelegatingResourceDescription();
+      description.addProperty("identifierType");
+      description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+      return description;
+  }
+
+  @Override
+  public DelegatingResourceDescription getUpdatableProperties() {
+      DelegatingResourceDescription description = new DelegatingResourceDescription();
+      description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+      return description;
+  }
+
+  @Override
+  public AutoGenerationOption newDelegate() {
+    return new AutoGenerationOption();
+  }
+
+  @Override
+  public AutoGenerationOption save(AutoGenerationOption autoGenerationOption) {
+    return Context.getService(IdentifierSourceService.class).saveAutoGenerationOption(autoGenerationOption);
+  }
+
+  @Override
+  public AutoGenerationOption getByUniqueId(String uniqueId) {
+    return Context.getService(IdentifierSourceService.class).getAutoGenerationOption(Integer.parseInt(uniqueId));
+  }
+
+  @Override
+  public void purge(AutoGenerationOption delegate, RequestContext context) throws ResponseException {
+    Context.getService(IdentifierSourceService.class).purgeAutoGenerationOption(delegate);
+  }
+
+  @Override
+  public void delete(AutoGenerationOption delegate, String id, RequestContext context) throws ResponseException {
+    throw new ResourceDoesNotSupportOperationException("Deleting of AutoGenerationOption is not supported");
+  }
+
+  // @Override
+  // protected PageableResult doGetAll(RequestContext context) throws ResponseException {
+	// 	return new AlreadyPaged<AutoGenerationOption>(Context.getService(IdentifierSourceService.class).getAutoGenerationOption(3), context);
```

Comment:
Why did you comment this out?

Reply:
I was trying to implement it but I noticed that there's no `getAllAutoGenerationOptions` method in the `IdentifierSourceService` class. The only option available requires passing  `PatientIdentifierType` as a parameter and I tried it and it was throwing error. I talked to @adamsdenniskariuki about it yesterday and I was hoping you'll guide me through doing it the right way.

Reply:
Please uncomment all code. At this stage, it is ok to have failing tests because am just reviewing to guide you in regards to how to fix them.

Reply:
Done!

Reply:
Have you created all the classes that the ticket description says?